src/routes/delegates/entities/delegate-params.entity.ts

```diff
@@ -0,0 +1,30 @@
+import { ApiPropertyOptional } from '@nestjs/swagger';
+
+export class DelegateParamsDto {
+  @ApiPropertyOptional()
+  safe?: string;
+  @ApiPropertyOptional()
+  delegate?: string;
+  @ApiPropertyOptional()
+  delegator?: string;
+  @ApiPropertyOptional()
+  label?: string;
+
+  constructor(
+    safe?: string,
+    delegate?: string,
+    delegator?: string,
+    label?: string,
+  ) {
+    this.safe = safe;
+    this.delegate = delegate;
+    this.delegator = delegator;
+    this.label = label;
+  }
+}
+
+export function isDelegateParamsDto(
+  dto: DelegateParamsDto,
+): dto is DelegateParamsDto {
+  return !!(dto.safe || dto.delegate || dto.delegator || dto.label);
+}
```

Comment:
I think this should not be used as a type predicate. If you look into `DelegateParamsDto` almost any TS object can matched with `DelegateParamsDto` because all the fields are nullable. So the question is not if it is a `DelegateParamsDto` but if at least one of the fields is not null.

So the following is fine:

``` typescript
if (
      !(
        delegateParams.safe ||
        delegateParams.delegate ||
        delegateParams.delegator ||
        delegateParams.label
      )
    ) {
      throw new HttpException(
        'At least one query param must be provided',
        HttpStatus.BAD_REQUEST,
      );
    }
```

Also sidenote: you have double negation `!!` in the original condition which is not needed.

Reply:
Ok, agree maybe the function could be called isValid... instead isType. But is ok. 
Regarding double negation I was aware, because should return `true` when was valid. 
```
> !(("") || ("") || (null) )
true
> !!(("Hello") || ("") || (null) )
true
```

Reply:
Done [bfc29e8c8c0a08242b33faf5fcedf88679d43664](https://github.com/5afe/safe-client-gateway-nest/pull/133/commits/bfc29e8c8c0a08242b33faf5fcedf88679d43664)